src/gui/gui-buffer.h

```diff
@@ -178,6 +178,7 @@ struct t_gui_buffer
     regex_t *text_search_regex_compiled; /* regex used to search            */
     int text_search_where;             /* search where? prefix and/or msg   */
     int text_search_found;             /* 1 if text found, otherwise 0      */
+    struct t_gui_line *text_search_start_line; /* starting line for search  */
```

Comment:
This variable must be moved in struct t_gui_window_scroll (in gui-window.h), because you can perform different search in the same buffer, if it is displayed in multiple windows at same time; that's why it must be stored in the window (with other scroll variables).
And if possible, please rebase all your commits into a single one, thanks!